src/languageModelTools/analyzeFileTool.ts

```diff
@@ -0,0 +1,66 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2025 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+'use strict';
+
+import * as vscode from 'vscode';
+import { SonarLintExtendedLanguageClient } from '../lsp/client';
+import { IssueService } from '../issue/issue';
+
+interface IAnalyzeFileParameters {
+  filePath: string;
+}
+
+export class AnalyzeFileTool implements vscode.LanguageModelTool<IAnalyzeFileParameters> {
+  public static readonly toolName = 'sonarqube_analyzeFile';
+  constructor(readonly client: SonarLintExtendedLanguageClient) {
+  }
+
+  async invoke(
+    options: vscode.LanguageModelToolInvocationOptions<IAnalyzeFileParameters>,
+    _token: vscode.CancellationToken
+  ) {
+    const params = options.input;
+
+    // Check that the folder is not using Connected Mode
+    const fileUri = vscode.Uri.file(params.filePath);
+    // Open file in the editor
+    const textDocument = await vscode.workspace.openTextDocument(fileUri);
```

Comment:
This is the cheapest and most straightforward approach for the tool. If we see that it is widely used, we can rethink the approach and have the analysis results returned directly from the tool.